### DISCUSSION TITLE
Build fails for 3.14.0-13.0.pre

### DISCUSSION THREAD

**Discussion by bf-viabill** [codemagic-ci-cd/codemagic-docs#2020] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Set `flutter: 3.14.0-13.0.pre` in `codemagic.yaml`.

### Expected results

Build

### Actual results

== Use Flutter version 3.14.0-13.0.pre ==

Build failed :|
Unable to find Flutter version 3.14.0-13.0.pre

### Build id (optional)

64f7387f056de8c1650ee798

**Comment by dtrdic** [comment]
hey @bf-viabill this is because that version is not installed on VM. You can check machine specs [here](https://docs.codemagic.io/specs/versions-macos/).

But, if you want to use that specific version, there is a Flutter Version Management (FVM) tool available that allows you to easily select the desired Flutter version specified in your FVM configuration file. You can download the FVM package from here https://fvm.app/docs/getting_started/overview/

Once you have FVM set up, you can use it in Codemagic by selecting the "from FVM configuration" option from the dropdown menu in the Build settings. This will ensure that your project is built using the Flutter version specified in your FVM configuration.

**Reply by bf-viabill** [reply]
Thanks for the answer.

What is the codemagic.yaml equivalent of selecting the option in the dropdown menu?

**Reply by bf-viabill** [reply]
Oh I found this in documentation https://docs.codemagic.io/yaml-quick-start/building-a-flutter-app/#setting-the-flutter-version

I guess this will do it.

**Reply by bf-viabill** [reply]
I still get the error:

```
== Parsing flutter version from FVM config ==

== Use Flutter version 3.14.0-13.0.pre ==

Build failed :|
Unable to find Flutter version 3.14.0-13.0.pre
```

Build id: 64f8515d74a300def90ee419

**Reply by dtrdic** [reply]
Hey @bf-viabill, if you try the command `fvm releases`, you will see all available Flutter SDK releases for installation. The latest available version is `3.14.0-0.2.pre`. Although the version `3.14.0-13.0.pre` that you mentioned is indeed available for download, it appears that it's currently not available for use.
<img width="317" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/a20f6a22-e945-4495-951e-52b133f7aa1d">

**Reply by bf-viabill** [reply, marked as answer]
It is available when you use `fvm install 3.14.0-13.0.pre`.

I've ended up disabling the flutter in the CI and installing it through FVM using a custom script.
